routes/projects.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const db = require('../data/helpers/projectModel');
+
+
+function validateBody(arr) {
+    return function(req, res, next) {
+        try {
+            arr.map( (i)=>{
+                console.log(i);
+                if(!(typeof req.body[i]) || req.body[i] === "")
+                    throw "invalid request";
+            });
+            next();
+        }
+        catch(e){
+            res.status(400).json({errorMessage: e});
+        }
+    }
+}
+
+const router = express.Router();
+router.get('/', async (req,res) => {
+    try {
+       let result = await db.get();
+        if(result.length === 0) throw "no projects availible";
+        res.status(200).json(result);
+    }
+    catch(e) {
+        await res.status(500).json({errorMessage: e });
+    }
+});
+
+router.get('/:id', async (req,res) => {
+    try {
+        result = await db.get(req.params.id);
+        if((result || result.length === 0)) throw "project not availible";
+        else res.status(200).json(result);
+    }
+    catch(e) {
+        let errorCode = ((e==="post not availible") ? 404 : 500);
+        res.status(errorCode).json({errorMessage: e });
+    }
+});
+
+router.post('/', async (req, res) => {
+    try {
+        result = await db.insert({
```

Comment:
Could be simplified by using db.insert(req.body)

Reply:
Actually, i do need result = await because db.insert returns a promise. and I return the result to the user. If I did not return the result to the user I could have gotten away with await db.insert.

Reply:
That's not what I meant, you do need the await, I was talking about the object that you created inside of the .insert() call, you could have passed only req.body simplifying the code and making it easier to read.

Reply:
passing the req.body would be really dangerous or could cause database problems if the body doesn't match right. I don't know what the user is doing, I wanted to make sure only certain items were getting through, and in the middleware i was [planning] on checking that they were strings.

I see your point but that was my thought process.